tests/ptp/tbc.go

```diff
@@ -0,0 +1,152 @@
+// Copyright 2023 Red Hat, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package ptp
+
+import (
+	"time"
+	"os"
+	"os/exec"
+	"bufio"
+	"regexp"
+	"strings"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	"github.com/redhat-partner-solutions/vse-sync-testsuite/pkg/images"
+	//"github.com/redhat-partner-solutions/vse-sync-testsuite/pkg/pods"
+
+	"github.com/redhat-partner-solutions/vse-sync-testsuite/pkg/clients"
+	"github.com/redhat-partner-solutions/vse-sync-testsuite/pkg/collectors/devices"
+	log "github.com/sirupsen/logrus"
+)
+
+const (
+	PTP_GROUP_LENGTH = 4
+)
+const (
+	PTP_METRICS = 4
+)
+
+const (
+	ptpLogsProcessed = "/tmp/data.csv"
+)
+
+const (
+	ptpLogs = "/tmp/data.log"
+)
+
+func Readln(r *bufio.Reader) (string, error) {
+  var (isPrefix bool = true
+       err error = nil
+       line, ln []byte
+      )
+  for isPrefix && err == nil {
+      line, isPrefix, err = r.ReadLine()
+      ln = append(ln, line...)
+  }
+  return string(ln),err
+}
+
+func decodePtp4line(tstamp string, sync_metrics string, processed_file *os.File ) error {
+	//log.Info(sync_metrics)
+	re := regexp.MustCompile(`-?\d+`)
+	ptp4l_sample_data := re.FindAllString(sync_metrics,-1)
+	//log.Info("ALL DATA before join ", ptp4l_sample_data)
+	if len(ptp4l_sample_data) == PTP_METRICS {
+		//write timestamp, offset with master, state, frequency, and path delay
+		ptp4l_all_sample_data := []string{tstamp,ptp4l_sample_data[0],ptp4l_sample_data[1],ptp4l_sample_data[2],ptp4l_sample_data[3]}
+		join := strings.Join(ptp4l_all_sample_data,",")
+		_, err := processed_file.WriteString(join + "\n")
+		if (err != nil){
+			return err
+		}
+	}
+	return nil
+}
+
+func parsePtpLogs(filename string, filename_dst string) error {
+	ptpfile, err := os.Open(filename)
+	if err != nil {
+		log.Panic("error problem reading file")
+	}
+	r := bufio.NewReader(ptpfile)
+	processed_file, err1 := os.OpenFile(filename_dst, os.O_CREATE|os.O_WRONLY, 0666)
+	if err1 != nil {
+		return err1
+	}
+	_, err2 := processed_file.WriteString("tstamp,phase,state,freq,delay,event" + "\n")
+	if (err2 != nil){
+		return err2
+	}
+	sample, err3 := Readln(r)
```

Comment:
don't number the errors. If they need to be identified differently then name them descriptively, if they don't then the Go practice is to reuse the same var.

Reply:
Fixed. Intended to use the same variable. No reason to use multiple variables that I can think of now.